appcelerator/titanium_mobile
# fix(ios): properly dismiss search controller after editing

- Properly dismiss search controller after editing
  - Only one search controller can be active at a time

###### NOTE: No idea why the delay is necessary, it's originally from [here](https://github.com/appcelerator/titanium_mobile/commit/4af828c01cc2fe4e20ac3c93604e670f2e7a8ddf#diff-1a2a8f1ea092b927374c5e1a188c5d1629165b4b736b79889496d15854969ba9R2030)

##### TEST CASE
1. Enter text into _Search A_, table should update with filtered results
2. Click _Toggle List_
3. Enter text into _Search B_, table should update with filtered results
```JS
const win = Titanium.UI.createWindow({ layout: 'vertical' });
const toggle = Ti.UI.createButton({ top: 80, title: 'Toggle List' })
const search_a = Ti.UI.createSearchBar({ hintText: 'Search A' });
const search_b = Ti.UI.createSearchBar({ hintText: 'Search B' });
const tableView_a = Ti.UI.createTableView({
    data: [ createSection(5, 'A') ],
    search: search_a,
    height: '40%'
});
const tableView_b = Ti.UI.createTableView({
    data: [ createSection(5, 'B') ],
    search: search_b,
    height: '40%',
    visible: false
});

toggle.addEventListener('click', e => {
    tableView_a.visible = tableView_b.visible;
    tableView_b.visible = !tableView_b.visible;

    if (!tableView_a.visible) {
        search_a.blur();
    }
    if (!tableView_b.visible) {
        search_b.blur();
    }
});

win.add([toggle, tableView_a, tableView_b]);
win.open();

function createSection(rows, suffix) {
    const section = Ti.UI.createTableViewSection({
        headerTitle: `Section ${suffix}`
    });

    for (let i = 1; i <= rows; i++) {
        section.add(Ti.UI.createTableViewRow({
            title: `Row #${i}`
        }));
    }

    return section;
}
```

[JIRA Ticket](https://jira.appcelerator.org/browse/TIMOB-28492)
